File: core/src/main/java/br/com/boitata/BoitataGame.java
package br.com.boitata;

import com.badlogic.gdx.ApplicationAdapter;
import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.graphics.GL20;
import com.badlogic.gdx.graphics.Texture;
import com.badlogic.gdx.graphics.g2d.SpriteBatch;

//public class BoitataGame extends ApplicationAdapter {
//	SpriteBatch batch;
//	Texture img;
//
//	@Override
//	public void create () {
//		setBatch(new SpriteBatch());
//		setImg(new Texture("badlogic.jpg"));
//	}
//
//	@Override
//	public void render () {
//		Gdx.gl.glClearColor(1, 0, 0, 1);
//		Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
//		getBatch().begin();
//		getBatch().draw(getImg(), 0, 0);
//		getBatch().end();
//	}
//
//	@Override
//	public void dispose () {
//		getBatch().dispose();
//		getImg().dispose();
//	}
//}
//

File: core/src/main/kotlin/br/com/boitata/BoitataGame.kt
package br.com.boitata

import br.com.boitata.screen.MainPage
import com.badlogic.gdx.Game
import com.badlogic.gdx.Gdx
import com.badlogic.gdx.Gdx.gl
import com.badlogic.gdx.graphics.GL20
import com.badlogic.gdx.graphics.OrthographicCamera
import com.badlogic.gdx.graphics.g2d.SpriteBatch
import com.badlogic.gdx.scenes.scene2d.Stage
import com.badlogic.gdx.utils.viewport.ScreenViewport

class BoitataGame : Game() {
    internal lateinit var stage: Stage

    override fun resize(width: Int, height: Int) {
        stage.viewport = ScreenViewport(OrthographicCamera(width.toFloat(), height.toFloat()))
        (stage.camera as OrthographicCamera).setToOrtho(false, width.toFloat(), height.toFloat())

        stage.viewport.update(width, height)
    }

    override fun create() {
        stage = Stage(ScreenViewport(), SpriteBatch())
        Gdx.input.inputProcessor = stage
        setScreen(MainPage(this))
    }

    override fun dispose() {
        clearScreen()
        screen.dispose()
        stage.dispose()
    }

    private fun clearScreen() {
        gl.glClearColor(0f, 0f, 0f, 1f)
        gl.glClear(GL20.GL_COLOR_BUFFER_BIT)
    }
}